### PR TITLE
[PowerPage][Copilot] Nl2Site and Nl2Page Service Integration for Site Create

### PR DESCRIPTION
**Pull request by amitjoshi438**
This pull request introduces new constants and services to handle NL2Page and NL2Site requests in the Power Pages chat participant feature. The most important changes include the addition of new constants, creation of the `getNL2PageData` and `getNL2SiteData` functions, and the integration of telemetry for error tracking and success events.

### New Constants:
* [`src/common/chat-participants/powerpages/PowerPagesChatParticipantConstants.ts`](diffhunk://#diff-0b8d35efc4455b9b6e562130d5c35e7d2f87c4bf9635332fbfe408f939d414edR45-R46): Added constants for NL2Page and NL2Site request statuses and error messages. [[1]](diffhunk://#diff-0b8d35efc4455b9b6e562130d5c35e7d2f87c4bf9635332fbfe408f939d414edR45-R46) [[2]](diffhunk://#diff-0b8d35efc4455b9b6e562130d5c35e7d2f87c4bf9635332fbfe408f939d414edL60-R65)

### New Services:
* [`src/common/chat-participants/powerpages/commands/create-site/Nl2PageService.ts`](diffhunk://#diff-cc3925d7f7ec7d629e18b6b8820e302729d85df809e105583de36d1eb305c631R1-R70): Created the `getNL2PageData` function to handle NL2Page requests, including constructing request bodies, making API calls, and sending telemetry events.
* [`src/common/chat-participants/powerpages/commands/create-site/Nl2SiteService.ts`](diffhunk://#diff-cbbe9a6cf2beaed245771d848d55b1b7c0a6e3243fd257446ac328ed31e5c682R1-R55): Created the `getNL2SiteData` function to handle NL2Site requests, including constructing request bodies, making API calls, and sending telemetry events.

### CODE REVIEW

File: src/common/chat-participants/powerpages/PowerPagesChatParticipantConstants.ts [hunk 1]
```diff
@@ -42,6 +42,8 @@ export const DISCLAIMER_MESSAGE = vscode.l10n.t('Make sure AI-generated content
 export const NO_PROMPT_MESSAGE = vscode.l10n.t('Hi! Power Pages lets you build secure, professional websites that you can quickly configure and publish across web browsers and devices.\n\nTo create your website, visit the [Power Pages](https://powerpages.microsoft.com/).\nReturn to this chat and @powerpages can help you write and edit your website code.');
 export const PAC_AUTH_INPUT = vscode.l10n.t("Checking for active auth profile...");
 export const COPILOT_NOT_RELEASED_MSG = vscode.l10n.t("@PowerPages is not yet available in your region.")
+export const NL2PAGE_REQUEST_FAILED = 'Request failed for page type:';
+export const NL2SITE_INVALID_RESPONSE = 'Invalid response structure';
```
Comment (tyaginidhi) [inline src/common/chat-participants/powerpages/PowerPagesChatParticipantConstants.ts:46]: Lets have seprate files for telemeytry and localization constants - this is going to grow in near future

File: src/common/chat-participants/powerpages/commands/create-site/Nl2PageService.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+export async function getNL2PageData(aibEndpoint: string, aibToken: string, userPrompt: string, siteName: string, sitePagesList: string[], sessionId: string, telemetry: ITelemetry) {
+    const constructRequestBody = (pageType: string) => ({
+        "crossGeoOptions": {
+            "enableCrossGeoCall": true
+        },
+        "question": `${userPrompt} - ${pageType} page`,
+        "context": {
+            "shouldCheckBlockList": false,
+            "sessionId": sessionId,
+            "scenario": "NL2Page",
```
Comment (tyaginidhi) [inline src/common/chat-participants/powerpages/commands/create-site/Nl2PageService.ts:18]: all this should be constants values

File: src/common/chat-participants/powerpages/commands/create-site/Nl2PageService.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            "subScenario": "GenerateNewPage",
+            "version": "V1",
+            "information": {
+                "scope": "Page",
+                "includeImages": true,
+                "pageType": pageType === 'FAQ' ? 'FAQ' : 'Home',
+                "title": siteName,
+                "pageName": pageType,
+                "colorNumber": 7, // Add a function to get a random number
```
Comment (tyaginidhi) [inline src/common/chat-participants/powerpages/commands/create-site/Nl2PageService.ts:27]: use math.random for this

File: src/common/chat-participants/powerpages/commands/create-site/Nl2PageService.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        }
+    });
+
+    const requests = sitePagesList.map(async pageType => {
+        const requestBody = constructRequestBody(pageType);
+
+        const requestInit: RequestInit = {
+            method: "POST",
+            headers: {
+                "Content-Type": "application/json",
```
Comment (tyaginidhi) [inline src/common/chat-participants/powerpages/commands/create-site/Nl2PageService.ts:40]: we have a common util function for getHeader - lets use that - already used in PPAPI service

File: src/common/chat-participants/powerpages/commands/create-site/Nl2SiteService.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+export async function getNL2SiteData(aibEndpoint: string, aibToken: string, userPrompt: string, sessionId: string, telemetry: ITelemetry) {
+    const requestBody = {
+        "crossGeoOptions": {
+            "enableCrossGeoCall": true
+        },
+        "question": userPrompt,
+        "context": {
+            "sessionId": sessionId,
+            "scenario": "NL2Site",
+            "subScenario": "GenerateNewSite",
```
Comment (tyaginidhi) [inline src/common/chat-participants/powerpages/commands/create-site/Nl2SiteService.ts:18]: include const here

File: src/common/chat-participants/powerpages/commands/create-site/Nl2SiteService.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+                "minPages": 7,
+                "maxPages": 7
+            }
+        }
+    };
+
+    const requestInit: RequestInit = {
+        method: "POST",
+        headers: {
+            "Content-Type": "application/json",
```
Comment (tyaginidhi) [inline src/common/chat-participants/powerpages/commands/create-site/Nl2SiteService.ts:31]: same as above

**Review by tyaginidhi — APPROVED** [review]

**Review by sanxsth — APPROVED** [review]

**Pull request merged**